### ISSUE TITLE
Font-Patcher fails when applied to Monaco font with Ligatures on macOS 10.14 beta 10

### ISSUE THREAD

**Issue by carlca** [ryanoasis/nerd-fonts#292]
macOS 10.14 Mojave beta 10
MacVim Custom Version 8.1.280 (151)
iTerm2 v3.2.0

I currently am using a version of Monaco.ttf which has had the Ligaturizer patcher applied to it. My initial attempts to use this failed with File Open errors, but eventually the author, @toxicfrog sent me a version of Monaco.ttf which worked.

Now I'm trying to use your `vim-devicons` plugin and I am coming up with the fact that I need to have a Nerd Font version of this font. So I ran this command...

`fontforge -script ./bin/font-patcher /Users/carlca/Library/Fonts/LigaMonacoNew.ttf`

but received the error message...

```
Copyright (c) 2000-2014 by George Williams. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Based on sources from 09:31 UTC 21-Mar-2018-D.
 Based on source from git with hash:
Warning: Mac string is a subset of the Windows string in the 'name' table
 for the Copyright string in the English (US) language.
The glyph named fi is mapped to U+F001.
But its name indicates it should be mapped to U+FB01.
The glyph named fl is mapped to U+F002.
But its name indicates it should be mapped to U+FB02.
No configfile given, skipping configfile related actions
The requested file, original-source.otf, does not exist
Traceback (most recent call last):
  File "./bin/font-patcher", line 761, in <module>
    symfont = fontforge.open(__dir__+"/src/glyphs/"+patch['Filename'])
EnvironmentError: Open failed
``` 

Any idea what's wrong?

I have attached the file in question.

[LigaMonacoNew.ttf.zip](https://github.com/ryanoasis/nerd-fonts/files/2368174/LigaMonacoNew.ttf.zip)

Another factor may be the fact that I am not certain that I have loaded the `vim-devicons` plugin correctly. I am using the Pathogen plugin manager, and following advice from the author @tpope I used the line...

`set rtp+=~/clone/VimBox/dotVim/bundle/vim-devicons-master`
after the
`call pathogen#runtime_append_all_bundles()`
line 

to try to ensure that vim-devicons was the last plugin loaded. It's not inconceivable that I may have got this wrong here, since I am not greatly skilled in the ways of Vim!

All I'm after, at the end of the day, is a pretty looking NERDTree with lots of whizzy file icons!

Cheers,
Carl Caulkett

**Comment by carlca** [comment]
Don't worry. Whenever you have a moment, that'll be fine!

**Comment by pianycist** [comment]
I get a similar error with the same traceback on Linux Mint 18.3 when attempting to patch Input Mono Light. 

```fontforge -script font-patcher Input-Regular_\(InputMono-Light\).ttf
Copyright (c) 2000-2014 by George Williams. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Based on sources from 00:15 UTC 31-Jul-2017-ML-D.
 Based on source from git with hash: b9149c13e8f9464fc21473f1f676b36a2130775d
The glyph named Tcommaaccent is mapped to U+021A.
  But its name indicates it should be mapped to U+0162.
The glyph named Delta is mapped to U+0394.
  But its name indicates it should be mapped to U+2206.
The glyph named Omega is mapped to U+03A9.
  But its name indicates it should be mapped to U+2126.
The glyph named tcommaaccent is mapped to U+021B.
  But its name indicates it should be mapped to U+0163.
The glyph named mu is mapped to U+03BC.
  But its name indicates it should be mapped to U+00B5.
No configfile given, skipping configfile related actions
The requested file, original-source.otf, does not exist
Traceback (most recent call last):
  File "font-patcher", line 761, in <module>
    symfont = fontforge.open(__dir__+"/src/glyphs/"+patch['Filename'])
EnvironmentError: Open failed```

**Comment by reaysawa** [comment]
Hey @carlca , I have recently also tripped on this. I went to investigate the line, which is:

https://github.com/ryanoasis/nerd-fonts/blob/master/font-patcher#L761

So if you download the release tarball, you ought to recreate a "src/glyphs" directory with the same files as

https://github.com/ryanoasis/nerd-fonts/tree/master/src/glyphs

as they don't seem to be included in the tagged source releases for some reason. I was able to manually patch my font after that.

@ryanoasis Would it be feasible to bundle this directory by default with the tarball releases? Or even provide a separate zip containing them for each release? I think it would be helpful in the future for people not knowing how to deal with this.

**Comment by ourfor** [comment]
↱ ∂:  ~/Code/nerd-fonts 
├─      master   INSERT                                                  ✔  8192  12:49:36 
↳ ß:fontforge -script ./font-patcher ./monaco.ttf
Copyright (c) 2000-2019. See AUTHORS for Contributors.
 License GPLv3+: GNU GPL version 3 or later <http://gnu.org/licenses/gpl.html>
 with many parts BSD <http://fontforge.org/license.html>. Please read LICENSE.
 Version: 20190801
 Based on sources from 11:52 UTC  3-Aug-2019-D.
Warning: Mac string is a subset of the Windows string in the 'name' table
 for the Copyright string in the English (US) language.
名为gacute 的字形被映射到 U+0123.
但它的名称表明它应映射到 U+01F5.
名为Scommaaccent 的字形被映射到 U+015E.
但它的名称表明它应映射到 U+0218.
名为scommaaccent 的字形被映射到 U+015F.
但它的名称表明它应映射到 U+0219.
Traceback (most recent call last):
  File "./font-patcher", line 895, in <module>
    main()
  File "./font-patcher", line 889, in main
    patcher = font_patcher()
  File "./font-patcher", line 54, in __init__
    self.sourceFont = fontforge.open(self.args.font)
OSError: Open failed

**Comment by ourfor** [comment]
OS version is `10.14.5 (18F132)`
[font file](https://raw.github.com/todylu/monaco.ttf/master/monaco.ttf)

**Comment by cseelus** [comment]
The solution @resolritter posted works fine. Issue could be closed.

**Comment by Finii** [comment]
Nowadays you use the `FontPatcher.zip` from the release page that contains everything you need.
Or the docker image.

@resolritter 
> Would it be feasible to bundle this directory by default with the tarball releases?

This is exactly what is available now, maybe it was your idea. :+1:
https://github.com/ryanoasis/nerd-fonts/releases/download/v2.3.3/FontPatcher.zip

Furthermore there is a PR to have the same for the bleeding-edge font-patcher: #1044

**Comment by Finii** [comment]
The ominous error message from the top has been fixed with this f97d729b34
